File: frontend/src/app/shared/service/admin/salary/SalaryAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {SalaryDto} from 'src/app/shared/model/salary/Salary.model';
import {SalaryCriteria} from 'src/app/shared/criteria/salary/SalaryCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class SalaryAdminService extends AbstractService<SalaryDto, SalaryCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): SalaryDto {
        return new SalaryDto();
    }

    public constrcutCriteria(): SalaryCriteria {
        return new SalaryCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/salary/';
    }
}

File: backend/src/main/java/ma/zs/alc/service/facade/admin/salary/SalaryAdminService.java
package ma.zs.alc.service.facade.admin.salary;

import java.util.List;
import ma.zs.alc.bean.core.salary.Salary;
import ma.zs.alc.dao.criteria.core.salary.SalaryCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface SalaryAdminService extends  IService<Salary,SalaryCriteria>  {

    List<Salary> findByProfId(Long id);
    int deleteByProfId(Long id);
    long countByProfRef(String ref);




}
